packages/hop-node/src/vault/AaveVault.ts

```diff
@@ -0,0 +1,293 @@
+    UI_INCENTIVE_DATA_PROVIDER: '0xD01ab9a6577E1D84F142e44D49380e23A340387d',
+  },
+  polygon: {
+    LENDING_POOL_ADDRESS_PROVIDER: '0xa97684ead0e402dc232d5a977953df7ecbab3cdb',
+    LENDING_POOL: '0x794a61358D6845594F94dc1DB02A252b5b4814aD',
+    WETH_GATEWAY: '0x9BdB5fcc80A49640c7872ac089Cc0e00A98451B6',
+    REPAY_WITH_COLLATERAL_ADAPTER: '0xA125561fca253f19eA93970534Bb0364ea74187a',
+    SWAP_COLLATERAL_ADAPTER: '0x301F221bc732907E2da2dbBFaA8F8F6847c170c3',
+    WALLET_BALANCE_PROVIDER: '0xBc790382B3686abffE4be14A030A96aC6154023a',
+    UI_POOL_DATA_PROVIDER: '0x8F1AD487C9413d7e81aB5B4E88B024Ae3b5637D0',
+    UI_INCENTIVE_DATA_PROVIDER: '0x05E309C97317d8abc0f7e78185FC966FfbD2CEC0',
+  }
+}
+
+// note: aave v3 on arbitirum, aave v2 on ethereum (v3 not available on ethereum)
+const tokenAddresses: Record<string, any> = {
+  USDC: {
+    arbitrum: {
+      token: '0xFF970A61A04b1cA14834A43f5dE4533eBDDB5CC8',
+      aToken: '0x625e7708f30ca75bfd92586e17077590c60eb4cd'
+    },
+    ethereum: {
+      token: '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48',
+      aToken: '0xbcca60bb61934080951369a648fb03df4f96263c'
+    }
+  },
+  USDT: {
+    arbitrum: {
+      token: '0xfd086bc7cd5c481dcc9c85ebe478a1c0b69fcbb9',
+      aToken: '0x6ab707aca953edaefbc4fd23ba73294241490620'
+    },
+    ethereum: {
+      token: '0xdac17f958d2ee523a2206206994597c13d831ec7',
+      aToken: '0x3ed3b47dd13ec9a98b44e6204a523e766b225811'
+    }
+  },
+  DAI: {
+    arbitrum: {
+      token: '0xda10009cbd5d07dd0cecc66161fc93d7c9000da1',
+      aToken: '0x82e64f49ed5ec1bc6e43dad4fc8af9bb3a2312ee'
+    },
+    ethereum: {
+      token: '0x6b175474e89094c44da98b954eedeac495271d0f',
+      aToken: '0x028171bca77440897b824ca71d1c56cac55b68a3'
+    }
+  },
+  ETH: {
+    arbitrum: {
+      token: NativeTokenAddress,
+      aToken: '0xe50fa9b3c56ffb159cb0fca61f5c9d750e8128c8'
+    },
+    ethereum: {
+      token: NativeTokenAddress,
+      aToken: '0x030ba81f1c18d280636f32af80b9aad02cf0854e'
+    }
+  },
+  MATIC: {
+    polygon: {
+      token: NativeTokenAddress,
+      aToken: '0x6d80113e533a2c0fe82eabd35f1875dcea89ea97'
+    }
+  }
+}
+
+export class AaveVault implements Vault {
+  chain: Chain
+  token: string
+  signer: any
+  provider: any
+  decimals: number
+  tokenAddress: string
+  aTokenAddress: string
+  pool: Pool
+
+  constructor (chain: Chain, token: string, signer: any) {
+    if (!tokenAddresses[token]) {
+      throw new Error('token is not supported')
+    }
+    if (!tokenAddresses[token][chain]) {
+      throw new Error('chain is not supported')
+    }
+    if (!aaveAddresses[chain]) {
+      throw new Error('chain is not supported')
+    }
+    this.chain = chain
+    this.token = token
+    this.tokenAddress = tokenAddresses[token][chain].token
+    this.aTokenAddress = tokenAddresses[token][chain].aToken
+    this.provider = getRpcProvider(chain)!
+    this.signer = signer.connect(this.provider)
+    this.decimals = getTokenDecimals(token)
```

Comment:
It looks like it does not matter, but do any of the `this.decimal` use-cases (which look like the parsing/formatting), use the `aToken` instead of the canonical token? If so, should we store `this.aTokenDecimals`?

Reply:
the aave tokens should always have the same decimals as their canonical token (similar to how hTokens and canonical tokens have same decimals in hop)